src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/NeuralNetStreamingExecutor.java

```diff
@@ -0,0 +1,216 @@
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by sam on 11/17/17.
+ */
+@CommandLineProgramProperties(
+        summary = NeuralNetStreamingExecutor.USAGE_SUMMARY,
+        oneLineSummary = NeuralNetStreamingExecutor.USAGE_ONE_LINE_SUMMARY,
+        programGroup = VariantEvaluationProgramGroup.class
+)
+
+public class NeuralNetStreamingExecutor extends VariantWalker {
+    private final static String NL = String.format("%n");
+    static final String USAGE_ONE_LINE_SUMMARY = "Apply 1d Convolutional Neural Net to filter annotated variants";
+    static final String USAGE_SUMMARY = "Annotate a VCF with scores from 1d Convolutional Neural Network (CNN)." +
+            "The CNN will look at the reference sequence and variant annotations to determine a Log Odds Score for each variant.";
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "Output file")
+    private File outputFile = null; // output file produced by Python code
+
+    @Argument(fullName = "architecture", shortName = "a", doc = "Neural Net architecture and weights hd5 file", optional = false)
+    private String architecture = null;
+
+    @Argument(fullName = "keep-info", shortName = "ki", doc = "Keep info fields in the vcf-like score file.", optional = true)
+    private boolean keepInfo = true;
+
+    @Argument(fullName = "python-batch-size", shortName = "pbs", doc = "Size of batches for python to do inference.", optional = true)
+    private int pythonBatchSize = 256;
+
+    @Argument(fullName = "python-sync-frequency", shortName = "psf", doc = "Size of data to queue for Python streaming.", optional = true)
+    private int pythonSyncFrequency = 512;
+
+    // Create the Python executor. This doesn't actually start the Python process, but verifies that
+    // the requestedPython executable exists and can be located.
+    final StreamingPythonScriptExecutor pythonExecutor = new StreamingPythonScriptExecutor(true);
+
+    private FileOutputStream fifoWriter;
+    private VariantContextWriter vcfWriter;
+    private AsynchronousStreamWriterService<String> asyncWriter = null;
+    private List<String> batchList = new ArrayList<>(pythonBatchSize);
+
+    private boolean noSamples = true;
+    private int curBatchSize = 0;
+    private boolean waitforBatchCompletion = false;
+
+
+    @Override
+    public boolean requiresReference(){
+        return true;
+    }
+
+    @Override
+    public void onTraversalStart() {
+        // setup the header fields
+        VCFHeader inputHeader = getHeaderForVariants();
+        final Set<VCFHeaderLine> inputHeaders = inputHeader.getMetaDataInSortedOrder();
+        final Set<VCFHeaderLine> hInfo = new HashSet<>(inputHeaders);
+        hInfo.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.CNN_1D_KEY));
+        VariantRecalibrationUtils.addVQSRStandardHeaderLines(hInfo);
```

Comment:
They are included below.  It is convenient to have the VQSR header fields for comparing the CNN with the VQSLODs, but this might not work for VCFs that haven't gone through VQSR.. I will check.